starfyre/router.py

```diff
@@ -0,0 +1,58 @@
+import os
+from pathlib import Path
+
+class Router:
+    """
+    A router that handles file-based routing.
+
+    This router parses the specified pages directory to automatically generate routes based on
+    the file names. Each file in the pages directory is treated as a separate route, and the
+    component name is derived from the file name. The routes and their corresponding component
+    names are stored in a dictionary.
+
+    Parameters:
+        pages_directory (str): The path to the directory containing the pages files.
+    Example:
+        pages_directory = "test_app/pages"
+        file_router = FileBasedRouter(pages_directory)
+    """
+
+    def __init__(self, pages_directory):
+        super().__init__()
+        self.pages_directory = pages_directory
+        self.generate_routes()
```

Comment:
One suggestion - we don't need to call this function here as you are explicitly calling it in `__main__.py`